plasma_framework/contracts/src/exits/payment/PaymentStandardExitable.sol

```diff
@@ -0,0 +1,153 @@
+     */
+    struct StartStandardExitData {
+        UtxoPosLib.UtxoPos utxoPos;
+        PaymentTransactionModel.Transaction outputTx;
+        PaymentOutputModel.Output output;
+        uint192 exitId;
+        bool isTxDeposit;
+        bytes txInclusionProof;
+        bytes rlpTx;
+        bytes32 txBlockRoot;
+        uint256 txBlockTimeStamp;
+    }
+
+    event ExitStarted(
+        address indexed owner,
+        uint192 exitId
+    );
+
+    constructor(address _framework) public {
+        framework = IPlasmaFramework(_framework);
+        isDeposit = IsDeposit.Predicate(framework.CHILD_BLOCK_INTERVAL());
+        exitableTimestampCalculator = ExitableTimestamp.Calculator(framework.minExitPeriod());
+    }
+
+    /**
+     * @notice Starts a standard withdrawal of a given output. Uses output-age priority.
+     * @dev requires the exiting UTXO's token to be added via 'addToken'
+     * @param _utxoPos Position of the exiting output.
+     * @param _outputTx RLP encoded transaction that created the exiting output.
+     * @param _outputTxInclusionProof A Merkle proof showing that the transaction was included.
+     */
+    function startStandardExit(
+        uint192 _utxoPos,
+        bytes calldata _outputTx,
+        bytes calldata _outputTxInclusionProof
+    )
+        external
+        payable
+        onlyWithValue(standardExitBond)
+    {
+        StartStandardExitData memory data = setupStartStandardExitData(_utxoPos, _outputTx, _outputTxInclusionProof);
+        verifyStartStandardExitData(data);
+        saveStandardExitData(data);
+        enqueueStandardExit(data);
+
+        emit ExitStarted(data.output.owner(), data.exitId);
+    }
+
+    /**
+    Private functions
+    */
+
+    function setupStartStandardExitData(
+        uint192 _utxoPos,
+        bytes memory _outputTx,
+        bytes memory _txInclusionProof
+    )
+        private
+        view
+        returns (StartStandardExitData memory)
+    {
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(_utxoPos);
+        PaymentTransactionModel.Transaction memory outputTx = PaymentTransactionModel.decode(_outputTx);
+        PaymentOutputModel.Output memory output = outputTx.outputs[utxoPos.outputIndex()];
+        bool isTxDeposit = isDeposit.test(utxoPos.blockNum());
+        uint192 exitId = ExitId.getStandardExitId(isTxDeposit, _outputTx, utxoPos);
+        (bytes32 root, uint256 blockTimestamp) = framework.blocks(utxoPos.blockNum());
+
+        return StartStandardExitData({
+            utxoPos: utxoPos,
+            outputTx: outputTx,
+            output: output,
+            exitId: exitId,
+            isTxDeposit: isTxDeposit,
+            txInclusionProof: _txInclusionProof,
+            rlpTx: _outputTx,
+            txBlockRoot: root,
+            txBlockTimeStamp: blockTimestamp
+        });
+    }
+
+    function verifyStartStandardExitData(StartStandardExitData memory data)
+        private
+        view
+    {
```

Comment:
Would it be better to have the 'cheaper' verifications first?
E.g. if calculating the inclusion proof was last, then any of the other requirements would take much less gas if not met.

Not sure if it's worth trying to optimise this though...

Reply:
That's a good idea. We can apply that easily.